File: Kadarkuti_Marton/nemes_tihamer_1214/nto_mancala/cpptest/nto_mancala.cpp
/*Nemes Tihamér Online 2. forduló - 2. feladat - Nomád játék
Kadarkuti Márton */

#include <iostream>
#include <vector>
using namespace std;

int main() {
    // bemenet
    int osszeg;
    int a;
    int b;
    int m; //megoldasok szama
    vector<int> x_megold;
    vector<int> y_megold;

    cin >> osszeg;
    cin >> a;
    cin >> b;

    for (int x = 0; x < 1+ osszeg/a; x++) {
        for (int y = 0; y < 1+ osszeg/b; y++) {
            if (a*x + b*y == osszeg) {
                m++;
                x_megold.push_back(x);
                y_megold.push_back(y);
            }
        }
    }
    cout << m << "\n";
    for (int i = 0; i<m; i++) {
        cout << x_megold[i] << " " << y_megold[i] << "\n";
    }

    return 0;
}

File: Kadarkuti_Marton/nemes_tihamer_1214/nto_mancala/nto_mancala.py
# Nemes Tihamér Online 2. forduló - 2. feladat - Nomád játék
# Kadarkuti Márton

# bemenet
osszeg:int = int(input().strip())
a:int = int(input().strip())
b:int = int(input().strip())
megoldasok:list[tuple[int]] = []

# a feladat megoldasai azok az (x,y) szamparok, melyre ax+by=osszeg teljesul
# x es y minimum 0, maximum pedig rendre osszeg//a es osszeg//b

for x in range(1+ osszeg//a):
    for y in range(1+ osszeg//b):
        if a*x + b*y == osszeg:
            megoldasok.append( (x,y) )

# kimenet
print(len(megoldasok))
for i in megoldasok:
    print(i[0],i[1])